File: toy/next-board/server/src/main/java/me/kzv/server/domain/Post.java
package me.kzv.server.domain;

import lombok.Builder;
import lombok.Getter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Entity
public class Post extends BaseEntity {

    @Id
    @GeneratedValue
    @Column(name = "post_id")
    private Long id;
    private String identifier;
    private String title;

    @Column(columnDefinition = "TEXT")
    private String body;

    private String subName;
    private String username;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sub_id")
    private Sub sub;


    @OneToMany(mappedBy = "post")
    private List<Comment> comments = new ArrayList<>();

    @OneToMany(mappedBy = "post")
    private List<Vote> votes = new ArrayList<>();


    protected Post() {
    }

    @Builder
    public Post(String identifier, String title, String body, String subName, String username, User user, Sub sub) {
        this.identifier = identifier;
        this.title = title;
        this.body = body;
        this.subName = subName;
        this.username = username;
        this.user = user;
        this.sub = sub;
    }
}

File: hello-kotlin/kotlin-board/src/main/kotlin/me/kzv/board/entity/Post.kt
package me.kzv.board.entity

import javax.persistence.*

@Entity
class Post private constructor(id: Long?, title: String, content: String, writer: String, likeCnt: Long, disLikeCnt: Long, userId: Long) : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_id")
    var id: Long? = id
    private set

    var title: String = title
        private set

    var content: String = content
        private set

    var writer: String = writer
        private set

    var likeCnt: Long = likeCnt
        private set

    var dislikeCnt: Long = disLikeCnt
        private set

    var userId: Long = userId

    constructor(title: String, content: String, writer: String, userId: Long): this(null, title, content, writer, 0, 0, userId)

    fun increaseLikeCnt() {
        this.likeCnt++
    }

    fun decreaseLikeCnt() {
        this.likeCnt--
    }

    fun increaseDislikeCnt() {
        this.dislikeCnt++
    }

    fun decreaseDislikeCnt() {
        this.dislikeCnt--
    }
}